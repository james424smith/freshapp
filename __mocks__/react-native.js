import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
