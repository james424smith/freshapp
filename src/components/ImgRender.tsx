import React from "react";
import { Image, ScrollView } from "react-native";
import { RenderType } from "../interfaces";
import styles from "../styles/more_screens/ImgRenderStyles";

type Props = { document: string; resourceType: RenderType; fileType: string };

const chooseAppropriateFile = (
  type: RenderType,
  document: string,
  fileType: string
) => {
  if (type === "file") {
    return { uri: `file://${document}` };
  } else {
    return { uri: `data:application/${fileType};base64,${document}` };
  }
};

const ImageRender = (props: Props) => {
  const { resourceType, document, fileType } = props;
  const img = chooseAppropriateFile(resourceType, document, fileType);

  return (
    <ScrollView
      maximumZoomScale={2.5}
      minimumZoomScale={1}
      style={styles.scrollViewStyles}
      contentContainerStyle={styles.scrollViewContainerStyles}
    >
      {img && <Image source={img} resizeMode={"contain"} style={styles.img} />}
    </ScrollView>
  );
};

export default ImageRender;
