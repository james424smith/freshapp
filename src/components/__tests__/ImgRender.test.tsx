import React from "react";
import ImgRender from "../ImgRender";
import { render, cleanup } from "@testing-library/react-native";

describe("ImgRender snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when resourceType is base64", () => {
    const { toJSON } = render(
      <ImgRender document={"1234"} fileType="jpg" resourceType={"base64"} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when resourceType is file", () => {
    const { toJSON } = render(
      <ImgRender document={"1234"} fileType="jpg" resourceType={"file"} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
