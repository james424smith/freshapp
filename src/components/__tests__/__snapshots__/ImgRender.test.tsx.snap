// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgRender snapshot test should render correct ui when resourceType is base64 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  maximumZoomScale={2.5}
  minimumZoomScale={1}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "data:application/jpg;base64,1234",
        }
      }
      style={
        Object {
          "flex": 1,
          "width": 750,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`ImgRender snapshot test should render correct ui when resourceType is file 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  maximumZoomScale={2.5}
  minimumZoomScale={1}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "file://1234",
        }
      }
      style={
        Object {
          "flex": 1,
          "width": 750,
        }
      }
    />
  </View>
</RCTScrollView>
`;
